src/simply/simply_window_stack.c

```diff
@@ -31,31 +31,57 @@ struct __attribute__((__packed__)) WindowShowPacket {
 
 typedef struct WindowSignalPacket WindowSignalPacket;
 
-struct __attribute__((__packed__)) WindowSignalPacket {
-  Packet packet;
-  uint32_t id;
-};
+#define WindowSignalPacketDef { \
+  Packet packet;                \
+  uint32_t id;                  \
+}
+
+struct __attribute__((__packed__)) WindowSignalPacket WindowSignalPacketDef;
 
 typedef WindowSignalPacket WindowHidePacket;
 
 typedef WindowHidePacket WindowEventPacket;
 
+struct __attribute__((__packed__)) WindowLoadEventPacket {
+  union {
+    struct WindowSignalPacketDef;
+  };
+  GSize size;
+};
+
+typedef struct WindowLoadEventPacket WindowLoadEventPacket;
+
 typedef WindowEventPacket WindowShowEventPacket;
 
 typedef WindowEventPacket WindowHideEventPacket;
 
 static bool s_broadcast_window = true;
 
-static bool send_window(SimplyMsg *self, Command type, uint32_t id) {
+static bool send_window_event_packet(SimplyMsg *self, WindowEventPacket* packet) {
   if (!s_broadcast_window) {
     return false;
   }
+  return simply_msg_send_packet(&packet->packet);
+}
+
+#define window_event_packet_field_init(name, packet_type, id) \
+  .packet.type = (packet_type), \
+  .packet.length = sizeof(name), \
+  .id = (id)
+
+static bool send_window(SimplyMsg *self, Command type, uint32_t id) {
   WindowEventPacket packet = {
-    .packet.type = type,
-    .packet.length = sizeof(packet),
-    .id = id,
+    window_event_packet_field_init(packet, type, id),
```

Comment:
You can create a static function `prv_init_window_event_packet` instead of a macro, and the compiler will semi-intelligently decide whether to inline the code or not (or we can give it attributes which way we'd want it to go). Static functions are to be prefixed with `prv_` but a majority of the code was written before this convention, so you can leave or change some of the older functions.